gluecodium/src/main/resources/cbridge_internal/include/WrapperCache.h

```diff
@@ -0,0 +1,72 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (C) 2016-2020 HERE Europe B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+// License-Filename: LICENSE
+//
+// -------------------------------------------------------------------------------------------------
+
+#pragma once
+
+#include <mutex>
+#include <unordered_map>
+
+namespace
+{
+class WrapperCache {
+public:
+    bool alive;
```

Comment:
According to the logic should be static? And i'd do it atomic (to decrease chance to receive cached value).
Honestly such approach hardly helps.

Reply:
Making it atomic is reasonable. Making it static is not necessary, since the class itself is not a "static class". The cache is a singleton, yes, but that's implemented through get_wrapper_cache() below.

Reply:
As for the approach itself, I don't like it either. I copied it from CachedProxyBase.h, where it was introduced by HARP team, supposedly to work around some static destruction order mishap. No idea why this never was a problem for other projects, but looks relatively harmless, so let's keep it.

Reply:
> Making it static is not necessary, since the class itself is not a "static class". The cache is a singleton, yes, but that's implemented through get_wrapper_cache() below.

May then extract variable from class? It's kind of UB when you use memory allocated for static class after destroying of the object.